### PR TITLE
Frontend: Migrate `Drawer.tsx` from aria-label e2e selectors to data-testid

### PR DESCRIPTION
**Pull request by codejagaban**
**What is this feature?**

This PR updates the e2e test labels on `packages/grafana-ui/src/components/Drawer/Drawer.tsx` to use `data-testid`

**Which issue(s) does this PR fix?**:

Related to #78412 

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -121,7 +122,8 @@ export function Drawer({
                   variant="secondary"
                   fill="text"
                   onClick={onClose}
-                  aria-label={selectors.components.Drawer.General.close}
+                  data-testid={selectors.components.Drawer.General.close}
+                  aria-label={t('selectors.components.Drawer.General.close', 'Close Drawer')}
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Drawer/Drawer.tsx:126]:
```suggestion
                  aria-label={t('grafana-ui.drawer.close', 'Close')}
```

And be sure to run `yarn i18n:extract` and commit the results :)

**Review by joshhunt — CHANGES_REQUESTED** [review]
Nice! Just need to improve the translations key, extract the phrase, and it's good to go!

File: packages/grafana-ui/src/components/Drawer/Drawer.tsx (hunk 1 above)
Comment (codejagaban) [reply packages/grafana-ui/src/components/Drawer/Drawer.tsx]: Hi @joshhunt  Changes implemented

**Review by joshhunt — APPROVED** [review]

**Review by joshhunt — CHANGES_REQUESTED** [review]
my bad - withdrawing approval until CI passes, but looks good so far.

**Review by joshhunt — APPROVED** [review]
lgtm! thanks for the contribution!

**Pull request merged**